### PR TITLE
test(textarea): Add tests for textarea view

### PR DESCRIPTION
**Pull request by mikelorant**
Add improved unit tests for textarea view function.

The end of buffer unit test was flawed and failed to properly check whether the end of buffer character was in the correct location. This test has been removed as this feature is now verified as part of these test cases.

### CODE REVIEW

**Comment by mikelorant** [comment]
@maaslalani @meowgorithm Can we please kick off the workflow checks.

**Comment by maaslalani** [comment]
> @maaslalani @meowgorithm Can we please kick off the workflow checks.

Workflow enabled!

**Comment by maaslalani** [comment]
@mikelorant, mind rebasing off master to fix the coverage workflow?

File: textarea/textarea_test.go [hunk 1]
```diff
@@ -416,15 +419,210 @@ func TestVerticalNavigationShouldRememberPositionWhileTraversing(t *testing.T) {
...
+	}{
+		{
+			name: "placeholder",
+			expected: heredoc.Doc(`
+				>   1 Hello, World!
+				>   ~
+				>   ~
+				>   ~
+				>   ~
+				>   ~
```
Comment (maaslalani) [inline textarea/textarea_test.go:436]:
```suggestion
				>   1 Hello, World!
				>  ~
				>  ~
				>  ~
				>  ~
				>  ~
```

Comment (maaslalani) [reply textarea/textarea_test.go:436]: The plan is to merge these tests then fix the current bug, then update the tests?

**Comment by maaslalani** [comment]
These tests look great, thank you so much for this PR. I really like the `heredoc` strings!

**Pull request merged**